### PR TITLE
[NFC] Update include paths in proxying tests

### PR DESCRIPTION
**Pull request by tlively**
Now that the proxying header is in a public location, these tests no longer need
to explicitly add the old private location to their include paths.

### CODE REVIEW

**Review by sbc100 — APPROVED** [review]

File: tests/test_core.py [hunk 1]
```diff
@@ -2578,34 +2578,30 @@ def test_pthread_proxying(self):
...
-                                 emcc_args=args)
+    self.do_run_in_out_file_test('pthread/test_pthread_proxying_dropped_work.c')
 
   @node_pthreads
   def test_pthread_proxying_refcount(self):
     self.set_setting('EXIT_RUNTIME')
     self.set_setting('PTHREAD_POOL_SIZE=1')
     self.set_setting('ASSERTIONS=0')
-    args = [f'-I{path_from_root("system/lib/pthread")}']
+    args = []
```
Comment (sbc100) [inline tests/test_core.py:2604]: You can just drop `args` here can't you?

Comment (sbc100) [reply tests/test_core.py:2604]: Oh.. I see not.

File: tests/test_core.py [hunk 2]
```diff
@@ -2578,34 +2578,30 @@ def test_pthread_proxying(self):
...
     if '-fsanitize=address' in self.emcc_args or '-fsanitize=leak' in self.emcc_args:
       args += ['-DSANITIZER']
```
[the remark this answers is not in the document]
Comment (tlively) [reply tests/test_core.py:2606]: No, unfortunately that does not cover lsan when it is used without asan.

[the remark this answers is not in the document]
Comment (sbc100) [reply tests/test_core.py:2606]: We use this elsewhere `#if __has_feature(leak_sanitizer) || __has_feature(address_sanitizer)` (see bind.h for example)

[the remark this answers is not in the document]
Comment (tlively) [reply tests/test_core.py:2606]: Oh wow, I could have sworn `__has_feature(leak_sanitizer)` didn't do anything, but testing it out I see that it does.

**Pull request merged**